src/sites/mod.rs

```diff
@@ -64,6 +65,7 @@ pub fn add_namespace(user: &GlobalUser, target: &mut Target, preview: bool) -> R
 pub fn directory_keys_values(
     target: &Target,
     directory: &Path,
+    exclude: Option<&HashSet<String>>,
```

Comment:
I think it would be nice to remove the `Option` - if callers don't want to ignore any keys, they can pass an empty set.
```suggestion
    exclude: &HashSet<String>,
```

Reply:
create a new HashSet and pass a ref to it, instead of None?

Reply:
Yup, exactly: `&HashSet::new()`

Reply:
I don't see the benefit, can you elaborate?

Reply:
`Option<HashSet>` has two ways to represent "don't exclude any files": `Some(HashSet::new())` and `None`. `HashSet` only has one.

Reply:
This also removes the `if let Some(...)` line below, which decreases the indentation a little.

Reply:
hmm. yea, i suppose. `None` is pretty clear from an intent standpoint. Going to leave as-is, and if its worthwhile we can revisit